src/main/java/org/broadinstitute/hellbender/tools/BwaMemIndexImageCreator.java

```diff
@@ -19,25 +23,51 @@
 )
 public final class BwaMemIndexImageCreator extends CommandLineProgram {
 
+    public static final String USE_EXISTING_INDEX_FULL_NAME = "useExistingIndexFilesIfAvailable";
+    public static final String USE_EXISTING_INDEX_SHORT_NAME = "useIndexFiles";
+
     @Argument(fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
-              shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME,
-              doc = "Input reference fasta file. The five bwa index files are assumed living in the same directory with the same prefix.")
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME,
+            doc = "Input reference fasta file. The five bwa index files are assumed living in the same directory with the same prefix.")
     private String referenceFastaLoc = null;
 
     @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
-              shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
-              doc = "Output reference index image file (ending in \".img\").",
-              optional = true)
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc = "Output reference index image file (ending in \".img\").",
+            optional = true)
     private String referenceIndexImageOutputLoc = null;
 
+    @Argument(fullName = USE_EXISTING_INDEX_FULL_NAME,
+            shortName = USE_EXISTING_INDEX_SHORT_NAME,
+            doc = "If there are some index files present, build the image out of those rather than the reference.",
+            optional = true)
+    private boolean useExistingIndexFilesIfAvailable = false;
+
     @Override
     protected final Object doWork() {
-
         if (referenceIndexImageOutputLoc == null) {
-            referenceIndexImageOutputLoc = referenceFastaLoc+".img";
+            referenceIndexImageOutputLoc = referenceFastaLoc + ".img";
         }
 
-        BwaMemIndex.createIndexImage(referenceFastaLoc, referenceIndexImageOutputLoc);
+        if (useExistingIndexFilesIfAvailable) {
+            if (checkIndexFilesArePresent(referenceFastaLoc)) {
+                BwaMemIndex.createIndexImageFromIndexFiles(referenceFastaLoc, referenceIndexImageOutputLoc);
+                return null;
+            } else {
+                logger.warn(String.format("User requested to use existing index files, " +
+                        "however no complete set of index file (%s.*) was found; " +
+                        "trying to create the image from the reference file directly", referenceFastaLoc));
```

Comment:
If the useExistingIndexFilesIfAvailable argument is advanced, I think that it is better to throw an `UserException` to make clear that it is not using previous indexes. I believe that if someone is asking for this option, he should be sure that the indexes are present (and generated with the same version or a compatible one).